Build broken with asnyc 111.13?

I just did an `opam update`, in the process pulling in `async` versino 111.13.00, and building ocaml-openflow now fails with:

```
ocaml setup.ml -build -j 4
Finished, 0 targets (0 cached) in 00:00:00.
+ ocamlfind ocamlc -c -g -I lib -syntax camlp4o -thread -package cstruct.syntax -package cstruct -package str -package sexplib.syntax -package cstruct.async -package packet -package async -package core -package threads -I async -I lib -o async/Async_OpenFlow_Platform.cmo async/Async_OpenFlow_Platform.ml
File "async/Async_OpenFlow_Platform.ml", line 97, characters 48-61:
Error: This expression has type
         ([> `Allow ] as 'a) Async_kernel.Deferred.t =
           'a Async_kernel.Ivar.Deferred.t
       but an expression was expected of type
         Impl.Client_id.t ->
         [ `Allow | `Deny of string option ] Async_extra.Import.Deferred.t
Command exited with code 2.
Compilation unsuccessful after building 82 targets (80 cached) in 00:00:00.
E: Failure("Command ''/home/basus/.opam/4.01.0/bin/ocamlbuild' lib/openflow.cma lib/openflow.cmxa lib/openflow.a lib/openflow.cmxs async/async.cma async/async.cmxa async/async.a async/async.cmxs examples/Learning_Switch.byte -use-ocamlfind -tag debug -j 4' terminated with error code 10")
make: *** [build] Error 1
```

I'm building the current HEAD of master and the build succeeds on a a different machine with `async` version 109.53.02.

--- a/async/Async_OpenFlow_Platform.ml
+++ b/async/Async_OpenFlow_Platform.ml
@@ -94,7 +94,7 @@ module Make(Message : Message) = struct
       ?log_disconnects
       ?buffer_age_limit ~port () =
     Impl.create ?max_pending_connections ?verbose ?log_disconnects
-      ?buffer_age_limit ~port ~auth:(fun _ _ -> return `Allow) ()
+      ?buffer_age_limit ~port ~auth:(fun _ _ _ -> return `Allow) ()
 
   let listen t =
     let open Impl.Server_read_result in

